config/confighttp/confighttp.go

```diff
@@ -259,6 +261,18 @@ func WithErrorHandler(e errorHandler) ToServerOption {
 
 // ToServer creates an http.Server from settings object.
 func (hss *HTTPServerSettings) ToServer(host component.Host, settings component.TelemetrySettings, handler http.Handler, opts ...ToServerOption) (*http.Server, error) {
+	if host, _, err := net.SplitHostPort(hss.Endpoint); err != nil {
```

Comment:
Can we not have http over unix sockets? (probably we don't support this).

Reply:
Listeners created through the `HTTPServerSettings` have the protocol hardcoded to `tcp`: https://github.com/open-telemetry/opentelemetry-collector/blob/bbf6043cc764bf73305a1d9c874c688f7aa4c1ef/config/confighttp/confighttp.go#L227

I guess someone could create a custom listener ignoring the configuration but create a server with the `ToServer` method. This seems clearly against the intention of the current API (TLS configuration and Endpoint from config would be ignored if not handled explicitly). Should we care about that case?

Reply:
If we care, I don't see a way out other than logging the error if `net.SplitHostPort` fails and skipping the unspecified check; we don't know within `ToServer` what kind of listener we will get.